catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -203,6 +222,40 @@ private void parseValidators(Map<String, Object> root, Outer outer) {
         outer.validators = validators;
     }
 
+    @SuppressWarnings("unchecked")
+    private void parseMetacardValidators(@Nullable Map<String, Object> root, Outer outer) {
+        if (root == null || root.get(METACARD_VALIDATORS_PROPERTY) == null) {
+            return;
+        }
+
+        List<MetacardValidatorDefinition> metacardValidators = new ArrayList<>();
+        Object metacardValidatorsObj = root.get(METACARD_VALIDATORS_PROPERTY);
+        if (metacardValidatorsObj instanceof List) {
+            List<Map<String, Object>> metacardValidatorsList =
+                    (List<Map<String, Object>>) metacardValidatorsObj;
+            for (Map<String, Object> metacardTypeList : metacardValidatorsList) {
```

Comment:
This has really high cyclomatic complexity. You might want to consider breaking some of these inner loops out to new methods. You can also invert the test on line 240 and simply `continue` if `validatorDefinitionObj` is not a `List`.

These changes aren't necessary, but might make this easier to read and easier to unit test.